### ISSUE TITLE
kops 1.6 unnecessarily requires subnets when creating instance groups

### ISSUE THREAD

**Issue by imsky** [kubernetes/kops#2656]
seeing this when i add a new instance group:

```
cannot create instance group without subnets; specify --subnet flag(s)
```

seems to be handled here: https://github.com/kubernetes/kops/blob/master/cmd/kops/create_ig.go#L133

this wasn't necessary before, and can be inferred from the cluster config itself. can the --subnet flag be made optional again?

**Comment by tuananh** [comment]
Can someone take a look at this?

**Comment by chrislovecnm** [comment]
kops 1.6?  We recommend that you use kops 1.7 with all k8s versions :)

k8s 1.4 -> 1.7 are supported by kops 1.7.

**Comment by imsky** [comment]
@chrislovecnm this issue was filed months ago, before 1.7. Reading the source, it still seems to be the case that specifying subnets is required when creating an instance group.

**Comment by chrislovecnm** [comment]
That is correct ... can we close this?

**Comment by imsky** [comment]
@chrislovecnm can the requirement to specify subnets be lifted? in pre-1.6 kops it wasn't required.

**Comment by imsky** [comment]
@chrislovecnm ping

```
$ kops version
Version 1.7.1 (git-c69b811)
$ kops create ig helloworld
Using cluster from kubectl context: example.com


cannot create instance group without subnets; specify --subnet flag(s)
```

**Comment by chrislovecnm** [comment]
It is a requirement for a subnet to specified at this time. How do you recommend handling this when we have 3+ subnets with a kops cluster?

**Comment by imsky** [comment]
@chrislovecnm if i recall correctly, the previous behavior was to add all cluster subnets to the instance group

this actually seems to be fixed in https://github.com/kubernetes/kops/commit/65bcbe7729c9f9ce92a445d05e160275f2e1c523 

now subnets will be inferred from the cluster spec here: https://github.com/kubernetes/kops/blob/7a2ff89d77a77667c556663db05557f6ada026cc/upup/pkg/fi/cloudup/populate_instancegroup_spec.go#L140-L143

**Comment by imsky** [comment]
still happens in 1.7.1:

```
$ kops version && kops create ig example
Version 1.7.1 (git-c69b811)
Using cluster from kubectl context: example.com


cannot create instance group without subnets; specify --subnet flag(s)
```

**Comment by imsky** [comment]
seems fixed in 1.10
